### PR TITLE
Add Admission Webhook Document

### PR DESCRIPTION
**Pull request by awgreene**
This commit introduces documentation that defines OLM support for
Validating and Mutating Admission Webhooks.

### CODE REVIEW

File: docs/admission-webhooks.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+OLM is capable of managing the lifecycle of [validating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and [mutating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) admission webhooks that are shipped alongside your operator.
+
+You may be unfamiliar with webhooks. At a high level, after a request has been authenticated and authorized, admission webhooks intercept requests against the Kubernetes API and have an opportunity to validate or update the object before it is saved in the object store. Please refer to the following table that highlights what each webhook is capable of:
+
+|                    | Validating Webhook | Mutating Webhook |
+|--------------------|--------------------|------------------|
+| Validating Objects |          x         |         x        |
+| Mutating Objects   |                    |         x        |
+
+If you are interested in learning more about admission webhooks, please review the [official kubeneretes documentation](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-are-they).
```
Comment (ecordell) [inline docs/admission-webhooks.md]:
```suggestion
If you are interested in learning more about admission webhooks, please review the [official kubernetes documentation](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-are-they).
```

File: docs/admission-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Defining your Webhook in the ClusterServiceVersion
+
+### Creating a Webhook
+
+It is outside the scope of this document to define the process of creating an admission webhook, however there are a few constraints that you should be aware of before developing your webhook if you plan to have it managed by OLM:
```
Comment (ecordell) [inline docs/admission-webhooks.md:20]: When we migrate this to the operator-framework site, we'll want links to sdk.

File: docs/admission-webhooks.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+#### Certificate Authority Restraints
```
Comment (ecordell) [inline docs/admission-webhooks.md]: Constraints?

File: docs/admission-webhooks.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+- Intercept requests that target all groups
+- Intercept requests that target the `operators.coreos.com` group
+- Intercept requests that target the `ValidatingWebhookConfigurations` or `MutatingWebhookConfigurations` resources
+
+### Defining a Webhook in a CSV
+
+OLM's ClusterServiceVersion resource was recently updated to include a [WebhookDefinition object](https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/clusterserviceversion_types.go#L164-L180) which can be used to define validating and mutating admission webhooks that will be shipped with the operator. For your convenience, the WebhookDefinition object can be seen below:
+
+```golang
```
Comment (ecordell) [inline docs/admission-webhooks.md]: I imagine for readers a yaml example would be more beneficial than the go struct

File: docs/admission-webhooks.md (hunk 3 above)
Comment (awgreene) [reply docs/admission-webhooks.md]: > While a constraint is a restriction or limitation on someone or something, a restraint is an act of holding back or keeping something or someone under control.

That seems fair, will update the doc.

File: docs/admission-webhooks.md (hunk 4 above)
Comment (awgreene) [reply docs/admission-webhooks.md]: Addressed in the latest version of the PR.

File: docs/admission-webhooks.md (hunk 3 above)
Comment (awgreene) [reply docs/admission-webhooks.md]: Addressed in the latest version of the PR.

File: docs/admission-webhooks.md (hunk 1 above)
Comment (awgreene) [reply docs/admission-webhooks.md]: Addressed in the latest version of the PR.

**Review by harishsurf — COMMENTED** [review]
Looks good. I felt the document could include a line or two on how to verify that the webhook is available on the cluster

**Comment by awgreene** [comment]
Closing this in favor of https://github.com/operator-framework/olm-docs/pull/17

**Pull request closed without merging**